### PR TITLE
CTCTRADERS-1577 Limit which customs office supervises x to UK-only of…

### PR DESCRIPTION
**Pull request by sawan07**
…fices

### CODE REVIEW

File: app/controllers/SimplifiedCustomsOfficeController.scala [hunk 1]
```diff
@@ -77,7 +77,7 @@ class SimplifiedCustomsOfficeController @Inject()(
                   consigneeName  = consigneeName.getOrElse(""),
                   customsOffice  = locationName,
                   form           = preparedForm,
-                  customsOffices = customsOffices,
+                  customsOffices = customsOffices.filter(customsOffice => customsOffice.id.startsWith("GB")),
```
Comment (PSouthern) [inline app/controllers/SimplifiedCustomsOfficeController.scala:80]: There are no specs for this controller change.  This should be added as part of the change to test the filter.

What happens if the data does not contain any offices with GB?  should we fall back to the full list?

**Review by PSouthern — CHANGES_REQUESTED** [review]

**Review by amaalali — CHANGES_REQUESTED** [review]
I don't think we should need to do this in memory (it's about 3000 odd offices), instead the API provides an endpoint that does the filtering

**Review by PSouthern — CHANGES_REQUESTED** [review]
@sawan07 Luke has confirmed this is also required for normal arrivals too, i.e. in https://github.com/hmrc/declare-transit-movement-arrival-frontend/blob/master/app/controllers/CustomsOfficeController.scala

File: app/controllers/SimplifiedCustomsOfficeController.scala (hunk 1 above)
Comment (sawan07) [reply app/controllers/SimplifiedCustomsOfficeController.scala]: specs added.

**Review by amaalali — APPROVED** [review]

**Pull request merged**